nmrfit.py

```diff
@@ -0,0 +1,342 @@
+        fitdata.__init__(self,*args,**kwargs)
+        self.symbol_list = [r'x',r'\xi'] # note that it must not be possible to find part of one of the later strings by searching for one of the earlier strings
+        self.gen_symbolic(r'xismax(C)')
+        return
+    def fitfunc_raw(self,p,C):
+        '''just the actual fit function to return the array y as a function of p and x'''
+        x = p[0]*C
+        xi = p[1]
+        snrmax = x/(x+3)
+        return xi*(1./3.+2./3.*snrmax)
+    def set_guess(self,**kwargs):
+        newdata = self.copy()
+        maxC = lambda x: x == self.getaxis(self.fit_axis).max()
+        self.guesses = {r'xi':newdata[self.fit_axis,maxC].data[-1],
+            'x':1.}
+        self.guesses.update(kwargs)
+        return
+    def guess(self):
+        r'''provide the guess for our parameters, which is specific to the type of function'''
+        if not hasattr(self,'guesses'):
+            self.set_guess()
+        return [self.guesses[x] for x in ['x','xi']] #guess [r'E_{max}',r'A']
+    def linfunc(self,x,y,xerr = None,yerr = None):
+        '''just the actual fit function to return the pair of arrays x',y' that should be linear
+        it accepts as inputs x and y, and it uses the output from the fit, where necessary
+        also optionally propagates the error based on yerr and xerr, which can be passed in to it
+        For the case of E_max, we want 1/(1-E) = 
+        '''
+        # note that there is some error associated with m(\infty) that I'm just taking for granted
+        #print "linfunc passed x=",x,"and y=",y
+        rety = 1./(y)
+        if yerr != None:
+            reterr = yerr/(y**2) # check this later
+        mask = isfinite(rety)
+        retx = 1./x # for instance, in emax, this is not just x
+        xname = r'1 / '+self.fit_axis # same as the fit axis
+        yname = r'$\frac{1}{1-E(p)}$'
+        #{{{ this should be pretty standardized
+        retval = nddata(rety,
+                [size(rety),1],
+                [xname,yname])
+        retval.labels([xname],
+                [retx.copy()])
+        if yerr != None:
+            retval.set_error(reterr)
+        #}}}
+        return retval
+#{{{ k_sigma s(p)
+class ksp (fitdata):
+    def __init__(self,*args,**kwargs):
+        fitdata.__init__(self,*args,**kwargs)
+        #self.symbol_list = [r'ks_{max}',r'p_{half}']
+        self.symbol_list = [r'ksmax',r'phalf']
+        self.starting_guesses = map(double,[r_[60,0.5],r_[20,0.05],r_[1.,1.]])
+        self.guess_lb = r_[0.1,1e-4]
+        self.guess_ub = r_[500.,30.]
+        self.zero_guess = r_[True,True]
+        self.gen_symbolic(r'k_{\sigma} s(p)')
+        return
+    def fitfunc_raw(self,p,x):
+        return p[0]/(p[1]+x)*x
+    def fitfunc_raw_symb(self,p,x):
+        try:
+            return p[0]/(p[1]+x)*x
+        except:
+            raise CustomError('types of p',p,map(type,p),'type of x',type(x),'x=',x)
+#}}}
+#}}}
```

Comment:
we don't want to commit this to pyspecdata

Reply:
agreed

Reply:
we do not want any of this, correct?

Reply:
no -- none if it -- if we want this for reference, I have this file in my notebook

Reply:
Alright - I just deleted it and pushed